client/packages/system/src/Encounter/DetailView/DetailView.tsx

```diff
@@ -183,6 +185,29 @@ export const DetailView: FC = () => {
     [data, setData]
   );
 
+  const onDelete = () => {
+    updateEncounter({ status: EncounterNodeStatus.Deleted });
+    setDeleteRequest(true);
+  };
+  useEffect(() => {
+    if (!deleteRequest) return;
+    if (
+      (data as Record<string, JsonData>)['status'] ===
+      EncounterNodeStatus.Deleted
+    ) {
+      saveData();
+      navigate(
+        RouteBuilder.create(AppRoute.Dispensary)
+          .addPart(AppRoute.Patients)
+          .addPart(encounter?.patient?.id ?? '')
+          .addQuery({
+            tab: PatientTabValue.Encounters,
+          })
+          .build()
+      );
```

Comment:
Currently, if there's an error, you are navigated away from the encounter page. It would be better to stay on the page, and only navigate if the delete succeeds:

```suggestion
      saveData().then(result => {
        if (!result) return;
        navigate(
          RouteBuilder.create(AppRoute.Dispensary)
            .addPart(AppRoute.Patients)
            .addPart(encounter?.patient?.id ?? '')
            .addQuery({
              tab: PatientTabValue.Encounters,
            })
            .build()
        );
      });
```